src/kv/untyped_map.h

```diff
@@ -141,7 +141,7 @@ namespace kv::untyped
         return committed_writes || change_set.has_writes();
       }
 
-      bool prepare() override
+      bool prepare(kv::Version& max_conflict_version) override
```

Comment:
Nit: AFAIA, the KV code so far doesn't use out parameters. Could `prepare()` return an `std::optional<kv::Version>` instead?

Reply:
Its not done often but in some places like commit_deserialised.

The version of this code where we return an optional is a fair bit messier which is why i went with the out param

Reply:
Same thing for hooks fwiw.